src/main/java/org/zalando/nakadi/metrics/MonitoringRequestFilter.java

```diff
@@ -34,11 +41,28 @@ public void doFilter(final ServletRequest request, final ServletResponse respons
         openHttpConnectionsCounter.inc();
         final Timer.Context timerContext = httpConnectionsTimer.time();
 
+        Timer.Context perPathTimerContext = null;
+        if (request instanceof HttpServletRequest) {
+            final HttpServletRequest httpRequest = (HttpServletRequest) request;
+            final String clientId = Optional.ofNullable(httpRequest.getUserPrincipal())
+                    .map(Principal::getName)
+                    .orElse("unauthenticated");
```

Comment:
for example, `/health` and `/metrics` endpoint. We might realise that they are called too often for example.